packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -53,14 +63,30 @@ class PouchManager {
 
     forEach(pouchPlugins, plugin => this.PouchDB.plugin(plugin))
     this.pouches = fromPairs(
-      this.doctypes.map(doctype => [
-        doctype,
-        new this.PouchDB(
-          getDatabaseName(this.options.prefix, doctype),
-          pouchOptions
-        )
-      ])
+      this.doctypes.map(doctype => {
+        const dbName = getDatabaseName(this.options.prefix, doctype)
+        return [
+          dbName,
+          new this.PouchDB(
+            getDatabaseName(this.options.prefix, doctype),
+            pouchOptions
+          )
+        ]
+      })
     )
+
+    Object.keys(this.pouches).forEach(dbName => {
+      const doctype = getDoctypeFromDatabaseName(dbName)
+      if (this.queryEngine === PouchDBQueryEngine) {
+        // Use existing pouch instance
+        const engine = new PouchDBQueryEngine(this, doctype)
+        engine.setDB(this.pouches[dbName])
```

Comment:
The `PouchDbQueryEngine` already retrieves the db when using `openDb`. So there should be no need to call `setDB` here.

So this can be replaced by:
```js
        const engine = new PouchDBQueryEngine(this, doctype)
        engine.openDB(dbName)
        this.dbQueryEngines.set(dbName, engine)
```
Which is nearly the same code in `setQueryEngine`:
```js
  setQueryEngine(name, doctype) {
        const engine = new this.options.queryEngine(this, doctype)
        engine.openDB(name)
        this.dbQueryEngines.set(name, engine)
        return engine
  }
```
And because you are in a `if (this.queryEngine === PouchDBQueryEngine)` then you can remove the if and mutualize the code here.

Reply:
Fair enough, I fixed it and moved it directly after the pouch initialization